EasyCommands/Common/Variables.cs

```diff
@@ -55,6 +55,34 @@ public class TernaryConditionVariable : Variable {
             public Primitive GetValue() => CastBoolean(condition.GetValue()) ? positiveValue.GetValue() : negativeValue.GetValue();
         }
 
+        public class VectorVariable : Variable {
+            public Variable vec, X, Y, Z;
+
+            public VectorVariable(Variable vector) {
+                vec = vector;
+            }
+            public VectorVariable(Variable x, Variable y, Variable z) {
+                X = x;
+                Y = y;
+                Z = z;
+            }
+
+            public Primitive GetValue() {
+                if (vec != null) {
+                    if (vec.GetValue().returnType == Return.VECTOR)
+                        return vec.GetValue();
+                    else
+                        throw new Exception("Invalid Vector variable.");
+                }
+                else {
+                    if (X.GetValue().returnType == Return.NUMERIC && Y.GetValue().returnType == Return.NUMERIC && Z.GetValue().returnType == Return.NUMERIC)
+                        return ResolvePrimitive(new Vector3D(CastNumber(X.GetValue()), CastNumber(Y.GetValue()), CastNumber(Z.GetValue())));
+                    else
+                        throw new Exception("Invalid Vector variable.");
+                }
```

Comment:
Let's de-scope VectorVariable to only handle the case of dynamically creating a Vector via a:b:c.

In the VectorVariable code I wouldn't do too much for checking the return types are numeric..  A simple CastNumber(v) should be sufficient.  Perhaps we can wrap it with a try catch to throw a clear "invalid variable used in vector creation" message or similar.

Reply:
`CastNumber is too lenient. If the user manages to stuff a vector into one component (which is possible with a few tricks) that gets resolved to its length and the evaluated vector would appaer valid.

Reply:
Fair enough I suppose.  I'm debating whether that's a bug or a feature.  

You can simplify I think:

if(NewList(x,y,z).TrueForAll(x => x.GetValue().returnType == Return.Numeric) return ResolvePrimitive( .... )
throw new Exception("Invalid Vector Variables");

Reply:
I would call it a bug.

Let me count the chars.
It is really difficult to fight my instinct to avoid allocations. 😄